Homeworks/module_6/hw_ig.py

```diff
@@ -0,0 +1,54 @@
+# 1. Implement a function that flatten incoming data:
+# non-iterables and elements from iterables (any nesting depth should be supported)
+# function should return an iterator (generator function)
+# don't use third-party libraries
+
+def merge_elems(*elems):
+    for elem in elems:
+        if hasattr(elem, "__iter__") and (hasattr(elem, "__next__") or not len(elem) == 1):
+            yield from merge_elems(*elem)
+        else:
+            yield elem
+
+
+# example input
```

Comment:
example input: [1]
output: [1] (list instead of list element)

+it's not necessary to check if the object has` __next__()`,` __iter__()` is enough

Reply:
string has __iter__ but doesn't have __next__. If I check only __iter__ then "maximum recursion depth exceeded"

but I updated it with other logic, should work now